[CI] AdJobRetryResilienceIT testJobReopensAfterIndexUnavailabilityDuringReassignment failing

**Build Scans:**
- [elasticsearch-intake #36569 / part4](https://gradle-enterprise.elastic.co/s/m6nvflz4aegvs)
- [elasticsearch-pull-request #132370 / part-4](https://gradle-enterprise.elastic.co/s/h4mc353xxp37g)
- [elasticsearch-pull-request #132367 / part-4-arm](https://gradle-enterprise.elastic.co/s/4mfnjin44ry5o)
- [elasticsearch-intake #36565 / part4](https://gradle-enterprise.elastic.co/s/y26fdkegjqqms)
- [elasticsearch-pull-request #132372 / part-4](https://gradle-enterprise.elastic.co/s/mzu2nx5nkoano)
- [elasticsearch-intake #36566 / part4](https://gradle-enterprise.elastic.co/s/qdmcbmk7u6nlu)
- [elasticsearch-pull-request #132096 / part-4](https://gradle-enterprise.elastic.co/s/aiaeob4xquz22)
- [elasticsearch-pull-request #132085 / part-4](https://gradle-enterprise.elastic.co/s/dp22qp4yyd5e2)
- [elasticsearch-pull-request #132065 / part-4](https://gradle-enterprise.elastic.co/s/rroqracjksm4c)
- [elasticsearch-pull-request #131773 / part-4](https://gradle-enterprise.elastic.co/s/k4zxzbadwfbr6)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:ml:internalClusterTest" --tests "org.elasticsearch.xpack.ml.integration.AdJobRetryResilienceIT.testJobReopensAfterIndexUnavailabilityDuringReassignment" -Dtests.seed=5EC4BEE63E07DB45 -Dtests.locale=nn-Latn-NO -Dtests.timezone=Pacific/Niue -Druntime.java=25
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.ml.integration.AdJobRetryResilienceIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testJobReopensAfterIndexUnavailabilityDuringReassignment'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: null
```

**Issue Reasons:**
- [main] 5 consecutive failures in test testJobReopensAfterIndexUnavailabilityDuringReassignment
- [main] 4 consecutive failures in step part4
- [main] 2 consecutive failures in step part-4
- [main] 4 consecutive failures in pipeline elasticsearch-intake
- [main] 3 consecutive failures in pipeline elasticsearch-pull-request
- [main] 26 failures in test testJobReopensAfterIndexUnavailabilityDuringReassignment (92.9% fail rate in 28 executions)
- [main] 4 failures in step part4 (100.0% fail rate in 4 executions)
- [main] 21 failures in step part-4 (91.3% fail rate in 23 executions)
- [main] 4 failures in pipeline elasticsearch-intake (100.0% fail rate in 4 executions)
- [main] 20 failures in pipeline elasticsearch-pull-request (95.2% fail rate in 21 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -354,9 +354,6 @@ tests:
 - class: org.elasticsearch.xpack.esql.heap_attack.HeapAttackSubqueryIT
   method: testGiantTextFieldInSubqueryIntermediateResults
   issue: https://github.com/elastic/elasticsearch/issues/144367
-- class: org.elasticsearch.xpack.ml.integration.AdJobRetryResilienceIT
-  method: testJobReopensAfterIndexUnavailabilityDuringReassignment
-  issue: https://github.com/elastic/elasticsearch/issues/144968
 - class: org.elasticsearch.xpack.esql.optimizer.rules.physical.local.ReorderLimitProjectAndOrderByGoldenTests
   method: testLimitByAndProjectNotSwapped
   issue: https://github.com/elastic/elasticsearch/issues/144978

--- a/x-pack/plugin/ml/src/internalClusterTest/java/org/elasticsearch/xpack/ml/integration/AdJobRetryResilienceIT.java
+++ b/x-pack/plugin/ml/src/internalClusterTest/java/org/elasticsearch/xpack/ml/integration/AdJobRetryResilienceIT.java
@@ -6,8 +6,10 @@
  */
 package org.elasticsearch.xpack.ml.integration;
 
+import org.elasticsearch.action.admin.cluster.reroute.ClusterRerouteUtils;
 import org.elasticsearch.action.admin.cluster.settings.ClusterUpdateSettingsRequest;
 import org.elasticsearch.cluster.node.DiscoveryNodeRole;
+import org.elasticsearch.cluster.routing.allocation.command.AllocateEmptyPrimaryAllocationCommand;
 import org.elasticsearch.common.settings.Settings;
 import org.elasticsearch.common.unit.ByteSizeValue;
 import org.elasticsearch.core.TimeValue;
@@ -156,10 +158,14 @@ public class AdJobRetryResilienceIT extends BaseMlIntegTestCase {
         );
         ensureStableCluster(3);
 
-        // Create indices: state/results on state-and-results, config on config
+        // Create indices: state/results on state-and-results, config on config.
+        // Explicit shard/replica settings override the randomized index template so that
+        // AllocateEmptyPrimaryAllocationCommand can reference shard 0 deterministically.
         indicesAdmin().prepareCreate(".ml-anomalies-shared-000001")
             .setSettings(
                 Settings.builder()
+                    .put("index.number_of_shards", 1)
+                    .put("index.number_of_replicas", 0)
                     .put("index.routing.allocation.include.ml-indices", "state-and-results")
                     .put("index.routing.allocation.exclude.ml-indices", "config,ml-only")
                     .build()
@@ -168,6 +174,8 @@ public class AdJobRetryResilienceIT extends BaseMlIntegTestCase {
         indicesAdmin().prepareCreate(".ml-state")
             .setSettings(
                 Settings.builder()
+                    .put("index.number_of_shards", 1)
+                    .put("index.number_of_replicas", 0)
                     .put("index.routing.allocation.include.ml-indices", "state-and-results")
                     .put("index.routing.allocation.exclude.ml-indices", "config,ml-only")
                     .build()
@@ -176,6 +184,8 @@ public class AdJobRetryResilienceIT extends BaseMlIntegTestCase {
         indicesAdmin().prepareCreate(".ml-config")
             .setSettings(
                 Settings.builder()
+                    .put("index.number_of_shards", 1)
+                    .put("index.number_of_replicas", 0)
                     .put("index.routing.allocation.exclude.ml-indices", "state-and-results")
                     .put("index.routing.allocation.include.ml-indices", "config")
                     .build()
@@ -208,13 +218,29 @@ public class AdJobRetryResilienceIT extends BaseMlIntegTestCase {
         internalCluster().stopNode(jobNode);
         ensureStableCluster(1, configNode);
 
-        // Config node's open attempts fail (state indices red). Allow allocation to config so shards relocate.
+        // Config node's open attempts fail (state indices red). Fix allocation filters and force-allocate
+        // the lost primaries on configNode. Two corrections are needed:
+        // 1. Clear the exclude filter (still blocked configNode) alongside updating include.
+        // 2. Allocate empty primaries because stateNode's shard data is permanently gone.
         indicesAdmin().prepareUpdateSettings(".ml-state", ".ml-anomalies-shared-000001")
-            .setSettings(Settings.builder().put("index.routing.allocation.include.ml-indices", "state-and-results,config").build())
+            .setSettings(
+                Settings.builder()
+                    .put("index.routing.allocation.include.ml-indices", "state-and-results,config")
+                    .putNull("index.routing.allocation.exclude.ml-indices")
+                    .build()
+            )
             .get();
 
-        // Wait for shards to allocate on the config node so GetJobsStats (used by awaitJobOpenedAndAssigned) can search the indices.
-        // Use a longer timeout than default ensureYellow (30s); on CI shard relocation after 2 node stops can take longer.
+        // Force-allocate empty primaries on configNode since the original primary data was on stateNode
+        // (which is now stopped). The ML job opening pipeline handles empty state gracefully: jobs with no
+        // model snapshot (newly created) skip snapshot verification and proceed to open directly.
+        ClusterRerouteUtils.reroute(
+            client(),
+            new AllocateEmptyPrimaryAllocationCommand(".ml-state", 0, configNode, true),
+            new AllocateEmptyPrimaryAllocationCommand(".ml-anomalies-shared-000001", 0, configNode, true)
+        );
+
+        // Wait for the allocated shards to initialise so GetJobsStats can search the indices.
         ensureGreen(TimeValue.timeValueMinutes(2), ".ml-state", ".ml-anomalies-shared-000001");
 
         awaitJobOpenedAndAssigned(jobId, null);

